spec/rspec/matchers/composable_spec.rb

```diff
@@ -12,6 +12,13 @@ module Matchers
         }.to fail_with(STDOUT.inspect)
       end
 
+      it "behaves correctly when surfacing descriptions from a readable IO object" do
+        allow(STDOUT).to receive(:each).and_yield
+        expect {
+          expect(3).to matcher_using_surface_descriptions_in(STDOUT)
+        }.to fail_with(STDOUT.to_s)
+      end
```

Comment:
This one was also ugly, because we can't get to the line in question without an instance of IO, and I didn't figure we'd want to open a fixture File for this spec (but if we do, that'd make it nicer).

Reply:
Would StringIO work?

Reply:
My first thought as well, but it's actually not a subclass of IO, just duck-typed like it! I assumed so hard I wrote the test and debugged it to find out I was wrong -.-